artichoke-backend/src/extn/core/time/args.rs

```diff
@@ -57,18 +57,52 @@ impl TryConvertMut<&mut [Value], Args> for Artichoke {
         for (i, &arg) in args.iter().enumerate() {
             match i {
                 0 => {
-                    let arg = to_int(self, arg)?;
-                    let arg: i64 = arg.try_convert_into(self)?;
+                    let arg: i64 = to_int(self, arg).and_then(|arg| arg.try_convert_into(self))?;
 
                     result.year = i32::try_from(arg).map_err(|_| ArgumentError::with_message("year out of range"))?;
                 }
                 1 => {
-                    // TODO: This should support 3 letter month names
-                    // as per the docs. https://ruby-doc.org/3.1.2/Time.html#method-c-new
-                    let arg = to_int(self, arg)?;
-                    let arg: i64 = arg.try_convert_into(self)?;
+                    // ```irb
+                    // 3.1.2 => Time.utc(2022, 2).month
+                    // => 2
+                    // 3.1.2 => class I; def to_int; 2; end; end
+                    // => :to_int
+                    // 3.1.2 => Time.utc(2022, I.new).month
+                    // => 2
+                    // 3.1.2 > Time.utc(2022, "feb").month
+                    // => 2
+                    // 3.1.2 > class A; def to_str; "feb"; end; end
+                    // => :to_str
+                    // 3.1.2 > Time.utc(2022, A.new).month
+                    // => 2
+                    // ```
+                    let month: i64 = if let Ok(month) = to_str(self, arg).and_then(|arg| {
+                        let month_str: &str = arg.try_convert_into_mut(self)?;
+                        match month_str {
+                            "jan" => Ok(1),
+                            "feb" => Ok(2),
+                            "mar" => Ok(3),
+                            "apr" => Ok(4),
+                            "may" => Ok(5),
+                            "jun" => Ok(6),
+                            "jul" => Ok(7),
+                            "aug" => Ok(8),
+                            "sep" => Ok(9),
+                            "oct" => Ok(10),
+                            "nov" => Ok(11),
+                            "dec" => Ok(12),
```

Comment:
Per the docs:

> A 3-character string that matches regular expression `/jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec/i`.

which seems to indicate this string is case insensitive. testing this out seems to confirm that:

```console
[3.1.2] > Time.utc(2022, "FEB").month
=> 2
[3.1.2] > Time.utc(2022, "FeB").month
=> 2
[3.1.2] > Time.utc(2022, "Feb").month
=> 2
[3.1.2] > Time.utc(2022, "fEB").month
=> 2
```

Reply:
I remember reading and swiftly ignoring the `/i` part of that regex since I was so focused on the values :sweat_smile:. Agree with the other comment though, comparing lowercase should fix it!